src/pages/Navigation.jsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { createStackNavigator, CardStyleInterpolators, TransitionPresets } from '@react-navigation/stack';
+import { createDrawerNavigator } from '@react-navigation/drawer';
+import { Ionicons } from '@expo/vector-icons';
+import GroupDrawer from './GroupDrawer';
+import MainPage from './MainPage';
+import LoginModal from './LoginModal';
+
+const Stack = createStackNavigator();
+const Drawer = createDrawerNavigator();
+
+function MainStack() {
+  return (
+    <Drawer.Navigator initialRouteName="Home" drawerContent={GroupDrawer}>
+      <Drawer.Screen
+        name="Home"
+        component={MainPage}
+        options={({ navigation }) => ({
+          headerShown: true,
+          title: 'ImageUs',
+          headerLeft: () => (<Ionicons name="ios-reorder-three-sharp" size={36} onPress={() => navigation.toggleDrawer()} />),
+        })}
```

Comment:
I think the icon needs a few px of left margin, it seems a little too close to the edge of the screen. Tested out `marginLeft: 16` and it seems to fit.